Schema/V1/no.ks.fiks.arkiv.v1.arkivering.arkivmelding.xsd

```diff
@@ -488,4 +488,15 @@
         </xs:sequence>
     </xs:complexType>
 
+    <xs:complexType name="referanseJournalpost">
+        <xs:sequence>
+            <xs:element name="systemID" type="n5mdk:systemID" minOccurs="0"/>
+            <xs:element name="journalaar" type="n5mdk:journalaar" minOccurs="0"/>
+            <xs:element name="journalsekvensnummer" type="n5mdk:journalsekvensnummer" minOccurs="0"/>
+            <xs:element name="journalpostnummer" type="n5mdk:journalpostnummer" minOccurs="0"/>
```

Comment:
Journalpostnummer er ikke unik. Den gir rekkefølge innenfor en saksmappe og må da ha saksaar og sakssekvensnummer i tillegg. Eksempel: 2022/15-3 der 3 er journalpostnummeret. Siden dette er en svært vanlig måte å referere en journalpost tror jeg det er greit å ta den inn. I arkivmeldingen er saksreferansen referanseForelderMappe som har saksnummer som complexType. Skulle vi hatt den sånn her må også journalaar/journalsekvensnummer være en complexType.

Reply:
Mener du noe sånt som dette?
```
<xs:complexType name="referanseJournalpost">
      <xs:sequence>
          <xs:element name="systemID" type="n5mdk:systemID" minOccurs="0"/>
          <xs:element name="journalAarSekvensnummerPostnummer" type="journalAarSekvensnummerPostnummer" minOccurs="0"/>
          <xs:element name="referanseEksternNoekkel" type="eksternNoekkel" minOccurs="0" maxOccurs="1"/>
          <xs:element name="registreringsID" type="n5mdk:registreringsID" minOccurs="0"/>
      </xs:sequence>
  </xs:complexType>

  <xs:complexType name="journalAarSekvensnummerPostnummer">
      <xs:sequence>
          <xs:element name="journalaar" type="n5mdk:journalaar" minOccurs="0"/>
          <xs:element name="journalsekvensnummer" type="n5mdk:journalsekvensnummer" minOccurs="0"/>
          <xs:element name="journalpostnummer" type="n5mdk:journalpostnummer" minOccurs="0"/>
      </xs:sequence>
  </xs:complexType>
```

Reply:
Nei, de to første er journalpostens journalnummer, fra gammelt av det nummeret som ble påført i det brevet ble ført inn i journalboken som danskekongen innførte. De vises typisk som journalnummer/journalår for å skille fra saksår/saksnummer (1234/2022). Saksnummeret sammen med journalpostnummeret er dagens mest vanlige måte å angi en journalpost på (2022/123-5).

Skal vi være konsekvent bør vi lage en complexType journalnummer som inneholder journalaar og journalsekvensnummer. Og så bør vi lage en extension til saksnummer som f.eks. heter saksJournalpostnummer og utvider med journalpostnummer. Disse to kan da benyttes som journalpostreferanser på linje med saksnummer som saksreferanse.

Ingen streng som må tolkes! Dette er separate felt og i forbindelse med søk kan det være aktuelt å angi de hver for seg.

Reply:
Slik mener du?
  ```
<xs:complexType name="referanseJournalpost">
        <xs:sequence>
            <xs:element name="systemID" type="n5mdk:systemID" minOccurs="0"/>
            <xs:element name="journalnummer" type="journalnummer" minOccurs="0"/>
            <xs:element name="journalpostnummer" type="n5mdk:journalpostnummer" minOccurs="0"/>
            <xs:element name="saksJournalpostnummer" type="saksJournalpostnummer" minOccurs="0"/>
            <xs:element name="referanseEksternNoekkel" type="eksternNoekkel" minOccurs="0" maxOccurs="1"/>
            <xs:element name="registreringsID" type="n5mdk:registreringsID" minOccurs="0"/>
        </xs:sequence>
    </xs:complexType>

    <xs:complexType name="journalnummer">
        <xs:sequence>
            <xs:element name="journalaar" type="n5mdk:journalaar" minOccurs="0"/>
            <xs:element name="journalsekvensnummer" type="n5mdk:journalsekvensnummer" minOccurs="0"/>
        </xs:sequence>
    </xs:complexType>

    <xs:complexType name="saksJournalpostnummer">
        <xs:complexContent>
            <xs:extension base="saksnummer">
                <xs:sequence>
                    <xs:element name="journalpostnummer" type="n5mdk:journalpostnummer" minOccurs="0"/>
                </xs:sequence>
            </xs:extension>
        </xs:complexContent>
    </xs:complexType>
```

Reply:
Journalpostnummer er nå duplisert. Det må tas bort fra referanseJournalpost og kun være via saksJournalpostnummer. Kan evt. vurdere om det skal være minOccurs 1 på enkeltelementene i journalnummer og saksjournalpostnummer. Det er det implisitt i saksnummer siden det ikke er minOccurs der.

Reply:
Da har jeg oppdatert koden i henhold til det vi kom fram til her. Tar du en titt?